### PR TITLE
Thomas/pet search user8/9

### PR DESCRIPTION
**Pull request by thawley2**
Added pet searching functionality to the application show page, to be able to search for a pet using a partial name and case-insensitive searching. 

Updated application show page with some logic to determine when to display the pet search results. 
Refactored the applications_controller to clean up the show action and move the logic for searching into a helper method 'search_by_name(name)'

### CODE REVIEW

**Review by Bluedevil667 — APPROVED** [review]
Nice job working out the .search method issue you were having! That was impressive!

File: app/controllers/applications_controller.rb [hunk 1]
```diff
@@ -3,17 +3,14 @@ class ApplicationsController < ApplicationController
   def show
     @application = Application.find(params[:id])
     @pets = @application.pets
-    @pets_search = Pet.where(name: params[:search])
-  end
-
-  def application_params
-    params.permit(:name, :address, :city, :state, :zip, :description)
+    @pets_search = search_by_name(params[:search])
```
Comment (thawley2) [inline app/controllers/applications_controller.rb:6]: I think I messed up and we should change the params[:search] to search_params.

File: app/controllers/applications_controller.rb [hunk 2]
```diff
@@ -23,12 +20,24 @@ def create
...
+  
   def update
     application = Application.find(params[:id])
     application.update(status: 1)
     application.update(good_owner: params[:good_owner])
     redirect_to "/applications/#{application.id}"
   end
+
+  def search_by_name(name)
+    Pet.search(name) if name.present?
```
Comment (thawley2) [inline app/controllers/applications_controller.rb:32]: if we change the above comment then we will have to change name to name[:search). But that might look weird if that line looks like `Pet.search(name[:search]) if name[:search].present?`

File: app/controllers/applications_controller.rb (hunk 1 above)
Comment (thawley2) [reply app/controllers/applications_controller.rb:6]: Now looking at the below comment, maybe we change it to `@pets_search = search_by_name(search_params[:search])`

**Review by thawley2 — COMMENTED** [review]
A thought for some later refactoring.

File: app/controllers/applications_controller.rb (hunk 1 above)
Comment (thawley2) [reply app/controllers/applications_controller.rb:6]: Well jeeze, now I think we could get rid of the search_by_name method all together and just have `@pets_search = Pets.search(search_params[:search])`

**Pull request merged**